lib/wallet/src/Cardano/Wallet/Shelley/Pools.hs

```diff
@@ -1032,6 +1020,60 @@ gcDelistedPools gcStatus tr DBLayer{..} fetchDelisted = forever $ do
     threadDelay sleepTime
     pure ()
 
+--------------------------------------------------------------------------------
+-- Types
+--------------------------------------------------------------------------------
+
+data StakePool = StakePool
```

Comment:
There is an argument to be made that the `ApiStakePool` type was a particular view on the data associated with a `StakePool`, but was not a "full" or "canonical" representation of _all_ data related to a `StakePool`. 🤔 (The data associated with a stake pool can be distinguished by trust, time and computational difficulty, and it does make sense to have separate types for these different aspects.)

However, I think you're right — introducing a type `StakePool` is a good idea, even in the case where it's just a starting point for further evolution. 🤓👍 For example, the canonical type for `cost` would be `Coin` and the canonical type for `margin` should be `Percentage`.

Reply:
💯